internal/api/fetcher/configapi.go

```diff
@@ -35,17 +42,47 @@ func (f *ConfigAPIFetcher) FetchAzureSEVSNPVersionList(ctx context.Context, atte
 
 // FetchAzureSEVSNPVersion fetches the version information from the config API.
 func (f *ConfigAPIFetcher) FetchAzureSEVSNPVersion(ctx context.Context, attestation configapi.AzureSEVSNPVersionGet) (configapi.AzureSEVSNPVersionGet, error) {
-	// TODO(elchead): follow-up PR for AB#3045 to check signature (sigstore.VerifySignature)
+	urlString, err := attestation.URL()
+	if err != nil {
+		return attestation, err
+	}
+	versionBytes, err := fetchBytesFromRawURL(ctx, urlString, f.httpc)
+	if err != nil {
+		return attestation, fmt.Errorf("fetch version %s: %w", attestation.Version, err)
+	}
+
```

Comment:
Right now you are fetching resource A, sig(A) and then verifying that A and sig(A) match. Then you are calling another function that returns B. You are assuming that A==B. Instead you should Marshal the result of `fetch`, and check that the signature matches.